src/coreclr/vm/threads.cpp

```diff
@@ -6047,11 +6047,13 @@ BOOL Thread::SetStackLimits(SetStackLimitScope scope)
     {
         m_CacheStackBase  = GetStackUpperBound();
         m_CacheStackLimit = GetStackLowerBound();
+#if !defined(TARGET_WASM) || !defined(_DEBUG) // stack can start at address 0 on wasm/emscripten and usually does in Debug builds
```

Comment:
```suggestion
#if !defined(TARGET_WASM) //  WASM-TODO: stack can start at address 0 on wasm/emscripten and usually does in Debug builds
```
We should keep this as TODO. I do not think the system is going to work well with `m_CacheStackLimit` set to 0.